pom.xml

```diff
@@ -1053,6 +1053,27 @@
                             </archive>
                         </configuration>
                     </plugin>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>build-helper-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>include-multirelease-sources-java9</id>
+                                <phase>generate-resources</phase>
+                                <goals>
+                                    <goal>add-resource</goal>
+                                </goals>
+                                <configuration>
+                                    <resources>
+                                        <resource>
+                                            <directory>${project.basedir}/src/main/java9</directory>
+                                            <targetPath>META-INF/versions/9</targetPath>
```

Comment:
Can you confirm that this would only add the sources to the source JAR and not the main artifact?

Reply:
Good catch @dmlloyd ! Of course you are right, with this configuration also MR jar file contained the java source files which is something we don't want.